packages/rpc-transport-http/src/json-parse-with-bigint.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * This function is a replacement for `JSON.parse` that can handle large integers by
+ * parsing them as BigInts. It will only transform integers that are too large to be
+ * represented as a regular JavaScript number.
+ */
+export function jsonParseWithLargeIntegersAsBigInts(json: string): unknown {
+    return JSON.parse(wrapLargeIntegers(json), (_, value) => {
+        return isWrappedBigInt(value) ? unwrapBigInt(value) : value;
+    });
+}
+
+/**
+ * This function takes a JSON string and wraps any large, unsafe integers in a special
+ * format that can be later recognized and unwrapped by a `JSON.parse` reviver.
+ */
+export function wrapLargeIntegers(json: string): string {
+    return transformUnquotedSegments(json, unquotedSegment => {
+        return unquotedSegment.replaceAll(/(-?\d+)/g, (_, value: string) => {
+            return Number.isSafeInteger(Number(value)) ? value : wrapBigInt(value);
+        });
+    });
+}
+
+/**
+ * This function takes a JSON string and transforms any unquoted segments using the provided
+ * `transform` function. This means we can be sure that our transformations will never occur
+ * inside a double quoted string — even if it contains escaped double quotes.
+ */
+function transformUnquotedSegments(json: string, transform: (value: string) => string): string {
+    /**
+     * This regex matches any part of a JSON string that isn't wrapped in double quotes.
+     *
+     * For instance, in the string `{"age":42,"name":"Alice \"The\" 2nd"}`, it would the
+     * following parts: `{`, `:42,`, `:`, `}`. Notice the whole "Alice \"The\" 2nd" string
+     * is not matched as it is wrapped in double quotes and contains escaped double quotes.
+     *
+     * The regex is composed of two parts:
+     *
+     *   1. The first part `^([^"]+)` matches any character until we reach the first double quote.
+     *   2. The second part `("(?:\\"|[^"])+")([^"]+)` matches any double quoted string that may
+     *      and any unquoted segment that follows it. To match a double quoted string, we use the
```

Comment:
nit: are you missing a word or line here? after "string that may".

Reply:
Oh yeah I must have deleted a comment line by mistake like an idiot. I've forgotten what I initially wrote now but I've updated the docs so it's a least a proper sentence. 😄